jax/_src/api.py

```diff
@@ -552,6 +554,12 @@ def __init__(self, executable, in_tree, out_tree, donate_argnums,
     self.donate_argnums = donate_argnums
     self._no_kwargs = no_kwargs
 
+  def compiler_ir(self):
+    return self._executable.xla_executable().hlo_modules()
```

Comment:
Comment to explain why this exists? Is it supposed to be the optimized IR?

Reply:
Good idea. Yes it is. Added a comment.